framework/src/base/MooseApp.C

```diff
@@ -410,6 +411,32 @@ MooseApp::meshOnly(std::string mesh_file_name)
 }
 
 void
+MooseApp::registerRecoverableData(std::string name)
+{
+  _recoverable_data.insert(name);
+}
+
+MooseSharedPointer<Backup>
+MooseApp::backup()
+{
+  FEProblem & fe_problem = static_cast<FEProblem &>(_executioner->problem());
+
+  RestartableDataIO rdio(fe_problem);
+
+  return rdio.createBackup();
+}
+
+void
+MooseApp::restore(MooseSharedPointer<Backup> backup)
+{
+  FEProblem & fe_problem = static_cast<FEProblem &>(_executioner->problem());
+
+  RestartableDataIO rdio(fe_problem);
+
+  return rdio.restoreBackup(backup);
```

Comment:
Interesting, explicit return of a void method here. I'm kind of surprised that there's not a warning